addons/website_sale_comparison/views/website_sale_comparison_template.xml

```diff
@@ -9,6 +9,14 @@
         </xpath>
     </template>
 
+    <template id="product_add_to_compare" name='Add to comparison in product page' inherit_id="website_sale.product" priority="8">
+        <xpath expr="//div[@id='o_wsale_cta_wrapper']" position="after">
+            <t t-set="categories" t-value="product.valid_product_template_attribute_line_ids._prepare_categories_for_display()"/>
+            <t t-set="product_variant_id" t-value="product._get_first_possible_variant_id()"/>
+            <button t-if="not is_view_active('website_sale_comparison.product_attributes_body') and product_variant_id and categories" type="button" role="button" class="d-none d-md-block btn btn-link px-0 o_add_compare_dyn" aria-label="Compare" t-att-data-product-product-id="product_variant_id" data-action="o_comparelist"><span class="fa fa-exchange me-2"></span>Compare</button>
```

Comment:
```suggestion
<t t-if="not is_view_active(...)">
            <t t-set="categories" t-value="product.valid_product_template_attribute_line_ids._prepare_categories_for_display()"/>
            <t t-set="product_variant_id" t-value="product._get_first_possible_variant_id()"/>
            <button t-if="product_variant_id and categories"
                            type="button"
                            role="button"
                            class="d-none d-md-block btn btn-link px-0 o_add_compare_dyn"
                            aria-label="Compare"
                            t-att-data-product-product-id="product_variant_id"
                            data-action="o_comparelist">
                                <span class="fa fa-exchange me-2"/>Compare
                      </button>
            </t>
```
1) Avoid evaluating (potentially costly) expressions when possible
2) let's indent the code as we add it here.
3) if view `website_sale_comparison.add_to_compare` is disabled, the `Compare` button shouldn't be shown on the product page (as it was done before dfd9da8aab37225c3a5785e2845d94e471b506a7) but you are not checking that here AFAICS.

Reply:
done thanks for the feedback

Reply:
Point 3 was not handled in my suggestion, but it should be handled as well ;)